qiskit/validation/validate.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Validators for Qiskit validated classes."""
+
+from marshmallow import ValidationError
+from marshmallow.validate import Validator
+
+
+class Or(Validator):
+    """Validate using a boolean "or" against a list of Validators.
+
+    This validator accepts a list of other ``Validators``, and returns True if
+    any of those validators pass.
+
+    Examples:
+        wheels = Integer(validate=Or(Equal(4), Equal(2)))
+    """
+
+    def __init__(self, validators):
+        """
+        Args:
+            validators (list[Validator]): list of Validators.
+        """
+        self.validators = validators
+
+    def __call__(self, value):
+        for validator in self.validators:
+            try:
+                return validator(value)
+            except ValidationError:
+                pass
+
+        raise ValidationError('nope')
+
+
+class PatternProperties(Validator):
```

Comment:
This validator seems to me like a kind of schema where some fields can be added dynamically if they follow some patterns. Also, notice that it is partially repeating what a schema does by trying to deserialize the values of each property which is the main job of a schema.

As mentioned before, since the usage of this is confined to modelling `counts`, I'd recommend to use it internally in the suggested alternative model of `counts`.